lib/uniware/client.rb

```diff
@@ -176,6 +176,24 @@ def create_pending_purchase_order(data, code)
       perform_operation("CreatePurchaseOrderRequest", body, facility_endpoint("#{code}"))
     end
 
+    def switch_sale_order_item_facility_request(data, code)
```

Comment:
Can we have 4 parameters to the method? Signature can be

def switch_sale_order_item_facility(order_code, item_codes, facility, facility_code)

Does that make sense? What does data['Facility'] contain?

The other option is to combine order_code & item_codes into an order hash with a code key & an items key.